sdks/go/pkg/beam/runners/dataflow/dataflow.go

```diff
@@ -51,26 +51,31 @@ import (
-	workerRegion         = flag.String("worker_region", "", "Dataflow worker region (optional)")
-	workerZone           = flag.String("worker_zone", "", "Dataflow worker zone (optional)")
+	endpoint               = flag.String("dataflow_endpoint", "", "Dataflow endpoint (optional).")
+	stagingLocation        = flag.String("staging_location", "", "GCS staging location (required).")
+	image                  = flag.String("worker_harness_container_image", "", "Worker harness container image (required).")
+	labels                 = flag.String("labels", "", "JSON-formatted map[string]string of job labels (optional).")
+	serviceAccountEmail    = flag.String("service_account_email", "", "Service account email (optional).")
+	numWorkers             = flag.Int64("num_workers", 0, "Number of workers (optional).")
+	workerHarnessThreads   = flag.Int64("number_of_worker_harness_threads", 0, "The number of threads per each worker harness process (optional).")
+	maxNumWorkers          = flag.Int64("max_num_workers", 0, "Maximum number of workers during scaling (optional).")
+	diskSizeGb             = flag.Int64("disk_size_gb", 0, "Size of root disk for VMs, in GB (optional).")
+	diskType               = flag.String("disk_type", "", "Type of root disk for VMs (optional).")
+	autoscalingAlgorithm   = flag.String("autoscaling_algorithm", "", "Autoscaling mode to use (optional).")
+	zone                   = flag.String("zone", "", "GCP zone (optional)")
+	kmsKey                 = flag.String("dataflow_kms_key", "", "The Cloud KMS key identifier used to encrypt data at rest (optional).")
+	network                = flag.String("network", "", "GCP network (optional)")
+	subnetwork             = flag.String("subnetwork", "", "GCP subnetwork (optional)")
+	noUsePublicIPs         = flag.Bool("no_use_public_ips", false, "Workers must not use public IP addresses (optional)")
+	tempLocation           = flag.String("temp_location", "", "Temp location (optional)")
+	machineType            = flag.String("worker_machine_type", "", "GCE machine type (optional)")
+	minCPUPlatform         = flag.String("min_cpu_platform", "", "GCE minimum cpu platform (optional)")
+	workerJar              = flag.String("dataflow_worker_jar", "", "Dataflow worker jar (optional)")
+	workerRegion           = flag.String("worker_region", "", "Dataflow worker region (optional)")
+	workerZone             = flag.String("worker_zone", "", "Dataflow worker zone (optional)")
+	dataflowServiceOptions = flag.String("dataflow_service_options", "", "Comma separated list of additional job modes and configurations (optional)")
+	flexRSGoal             = flag.String("flexrs_goal", "", "Which Flexible Resource Scheduling mode to run in (optional)")
+	enableHotKeyLogging    = flag.Bool("enable_hot_key_logging", false, "Specifies that when a hot key is detected in the pipeline, the literal, human-readable key is printed in the user's Cloud Logging project (optional).")
```

Comment:
Filed https://issues.apache.org/jira/browse/BEAM-14512 about below. We should comment this out for now and link to this JIRA. We can keep the rest of the plumbing though, as we'll need it later.

We can't support this yet because we don't support the "ToString" URN/Capability yet. It would be very similar to Drain, but without the user customization. Basically we add an exec Node that takes an element and uses fmt.Sprintf("%v") to turn it into a string. The element comes from a DataSource as usual, and is returned to the service with a DataSink, so it really is just "understand the URN" and "include it in the Capabilities list" (statically, all the time even).

https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto#L329

Unfortunately, this one is annoying to test because there's no closed form test runner for it. It's not hard to write a job that HotKeys at least, and that we're printing whatever the `String()` method provides (as handled by the `fmt` package).

Reply:
Done!